students/cfarris/lessons/lesson05/assignment/database.py

```diff
@@ -0,0 +1,167 @@
+#!/usr/bin/env Python3
+
+"""
+This set up is only for development use. Not production
+Uses Pymongo to access MongoDB database.
+Imports 3 files, directs them to different collections
+Retrieves information necessary for HP Norton collection
+
+Can be used as a module or stand-alone
+Lacks error checking of data (due to time constraints.)
+"""
+import csv
+from pymongo import MongoClient
+
+
+
+class MongoDBConnection():
+    """Connect to MongoDB"""
+    def __init__(self, host='127.0.0.1', port=27017):
+
+        self.host = host
+        self.port = port
+        self.connection = None
+
+    def __enter__(self):
+        self.connection = MongoClient(self.host, self.port)
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        self.connection.close()
+
+
+def print_mdb_collection(db, collection_name):
+    """Prints the collection specified"""
+    for doc in collection_name.find():
+        print(doc)
+
+
+def import_data(db, coll_dict):
+    """Adds files to collection. if the file/date type (k) fits into three categories
+       it will be added to that collection.
+    """
+    for k, v in coll_dict.items():
+        if k == 'rentals':
+            rentals_db = db['rentals_db']
+            rentals_data = read_csv(v)
+            result = rentals_db.insert_many(rentals_data)
+
+        elif k == 'customers':
+            customers_db = db['customers_db']
+            customers = read_csv(v)
+            result = customers_db.insert_many(customers)
+
+        elif k == 'products':
+            products_db = db['products_db']
+            products = read_csv(v)
+            result = products_db.insert_many(products)
+
+    return db.rentals_db.count(), db.customers_db.count(), db.products_db.count()
+
+
+def read_csv(v):
+    '''
+    Opens csv files, creates a list of dictionarys with matching keys from header to values
+    returns the dictionary and saves it to different mongodb collections based upon the
+    file name given to main.
+    '''
+    with open(v, 'r', encoding='utf-8-sig') as csv_file:
+        csv_reader = csv.reader(csv_file, delimiter=',')
+        header = True
+        first_line = []
+        coll_of_dicts = []
+        for row in csv_reader:
+            if header:
+                header = False
+                first_line = [value.strip() for value in row]
+            else:
+                new_dict = {}
+                str_row = []
+                str_row = [value.strip() for value in row]
```

Comment:
You don't need to initialize str_row with an empty list if you are doing a list comprehension. Nice comprehension.